Objective-C/CBLDocument.mm

```diff
@@ -407,48 +357,34 @@ - (BOOL) mergeWithConflictResolver: (id<CBLConflictResolver>)resolver
                           deletion: (bool)deletion
                              error: (NSError**)outError
 {
+    if (!resolver)
+        return convertError({LiteCoreDomain, kC4ErrorConflict}, outError);
+
     // Read the current revision from the database:
-    C4Document* rawDoc = [self readC4Doc_mustExist: YES error: outError];
-    if (!rawDoc)
+    auto database = self.database;
+    CBLReadOnlyDocument* current = [[CBLReadOnlyDocument alloc] initWithDatabase: database
+                                                                      documentID: self.documentID
+                                                                       mustExist: YES
+                                                                           error: outError];
+    if (!current)
         return NO;
-    
-    FLDict curRoot = nullptr;
-    auto curBody = rawDoc->selectedRev.body;
-    if (curBody.size > 0)
-        curRoot = (FLDict) FLValue_FromTrustedData({curBody.buf, curBody.size});
-    
-    // Create the current readonly document with the current revision:
-    CBLC4Document* curC4doc = [CBLC4Document document: rawDoc];
-    CBLFLDict* curDict = [[CBLFLDict alloc] initWithDict: curRoot
-                                                   c4doc: curC4doc
-                                                database: _database];
-    CBLReadOnlyDocument* current = [[CBLReadOnlyDocument alloc] initWithDocumentID: self.documentID
-                                                                             c4Doc: curC4doc
-                                                                        fleeceData: curDict];
+    CBLC4Document* curC4doc = current.c4Doc;
+
     // Resolve conflict:
     CBLReadOnlyDocument* resolved;
     if (deletion) {
         // Deletion always loses a conflict:
         resolved = current;
-    } else if (resolver) {
-        // Call the custom conflict resolver:
-        CBLReadOnlyDocument* base = [[CBLReadOnlyDocument alloc] initWithDocumentID: self.documentID
-                                                                              c4Doc: super.c4Doc
-                                                                         fleeceData: super.data];
+    } else {
```

Comment:
Actually this is not fully related to the PR so I could add this logic after merging the PR.